src/Sylius/Bundle/UserBundle/Controller/UserController.php

```diff
@@ -62,22 +63,26 @@ public function changePasswordAction(Request $request)
 
     public function requestPasswordResetTokenAction(Request $request)
     {
-        $generator = $this->container->get('sylius.user.token_provider');
+        $generator = $this->container->get('sylius.user.password_reset.token_provider');
 
         return $this->prepareResetPasswordRequest($request, $generator, UserEvents::REQUEST_RESET_PASSWORD_TOKEN);
     }
 
     public function requestPasswordResetPinAction(Request $request)
     {
-        $generator = $this->container->get('sylius.user.pin_provider');
+        $generator = $this->container->get('sylius.user.password_reset.pin_provider');
 
         return $this->prepareResetPasswordRequest($request, $generator, UserEvents::REQUEST_RESET_PASSWORD_PIN);
     }
 
     public function resetPasswordAction(Request $request, $token)
     {
         $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
-        $user = $this->findUserByToken($token);
+        /** @var UserInterface $user */
+        $user = $this->repository->findOneBy(['passwordResetToken' => $token]);
+        if (null === $user) {
```

Comment:
IIRC this would generate a 500 error, but we want to provide 404, or am I wrong?